srml/balances/src/lib.rs

```diff
@@ -437,28 +489,69 @@ where
 		<ReservedBalance<T, I>>::get(who)
 	}
 
-	fn slash(who: &T::AccountId, value: Self::Balance) -> Option<Self::Balance> {
+	fn withdraw<S: OnUnbalancedDecrease<Self::Balance>>(
+		who: &T::AccountId,
+		value: Self::Balance,
+		reason: WithdrawReason,
+		liveness: ExistenceRequirement,
+	) -> result::Result<(), &'static str> {
+		if let Some(new_balance) = Self::free_balance(who).checked_sub(&value) {
+			if liveness == ExistenceRequirement::KeepAlive && new_balance < Self::existential_deposit() {
+				return Err("payment would kill account")
+			}
+			Self::ensure_account_can_withdraw(who, value, reason, new_balance)?;
+			S::on_unbalanced_decrease(value)?;
+			Self::set_free_balance(who, new_balance);
+			Ok(())
+		} else {
+			Err("too few free funds in account")
+		}
+	}
+
+	fn slash<S: OnUnbalancedDecrease<Self::Balance>>(
+		who: &T::AccountId,
+		value: Self::Balance
+	) -> Option<Self::Balance> {
 		let free_balance = Self::free_balance(who);
 		let free_slash = cmp::min(free_balance, value);
 		Self::set_free_balance(who, free_balance - free_slash);
-		Self::decrease_total_stake_by(free_slash);
-		if free_slash < value {
-			Self::slash_reserved(who, value - free_slash)
+		let res = if free_slash < value {
```

Comment:
Could be:
```
if let Some(remaining_slash) = value.checked_sub(free_slash) {
 ...
}
```

Reply:
this is about to be rewritten in an upcoming PR anyway so will leave it until then